packages/core/src/createStore.ts

```diff
@@ -28,25 +32,31 @@ export function createStore(atom: Atom<any>, initState?: State): Store
 // for prevent using `delete` operator
 // (need perf tests)
 export function createStore(
-  atom: Atom<any> | State = defaultAtom,
-  initState: State = {},
+  atom?: Atom<any> | State,
+  initState?: State,
 ): Store {
-  if (!getIsAtom(atom)) {
-    initState = atom
-    atom = defaultAtom
-  }
   let atomsListeners: Map<TreeId, Function[]> = new Map<TreeId, Function[]>()
   let nextAtomsListeners: Map<TreeId, Function[]> = atomsListeners
   let actionsListeners: Function[] = []
   let nextActionsListeners: Function[] = actionsListeners
-  // const storeAtom = map('store', atom || defaultAtom, value => value)
+  let initialAtoms = new Set<TreeId>()
+  const state: State = {}
   const storeTree = new Tree('store')
-  storeTree.union(getTree(atom as Atom<any>)!)
-  const ctx = createCtx(initState, initAction)
-  storeTree.forEach(initAction.type, ctx)
-  const initialAtoms = new Set(Object.keys(ctx.stateNew))
-  // preloadedState needed to save data of lazy atoms
-  const state = assign({}, initState, ctx.stateNew) as State
+  if (atom !== undefined) {
+    if (typeof atom === 'object' && initState === undefined) assign(state, atom)
+    else {
+      if (!getIsAtom(atom)) throwError('Invalid atom')
+      if (typeof initState === 'object' && initState !== null)
+        assign(state, initState)
+      else if (initState !== undefined) throwError('Invalid initial state')
+
+      storeTree.union(getTree(atom as Atom<any>))
+      const ctx = createCtx(state, initAction)
+      storeTree.forEach(initAction.type, ctx)
+      assign(state, ctx.stateNew)
+      initialAtoms = new Set(Object.keys(ctx.stateNew))
+    }
+  }
```

Comment:
I think there can be more simple than tons of 'if'

Reply:
let propose a changes in a new PR